### PR TITLE
Add ZNCLDJ01LM Aqara C3 Curtain Motor

### PR DESCRIPTION
**Pull request by marthubner**
Adding support for Aqara C3 Curtain Motor based on my external converter presented [here](https://github.com/Koenkk/zigbee2mqtt/issues/19555#issuecomment-2269053169).

### CODE REVIEW

File: src/lib/lumi.ts [hunk 1]
```diff
@@ -2829,6 +2830,136 @@ export const fromZigbee = {
             }
         },
     } satisfies Fz.Converter,
+    lumi_curtain_speed: {
```
Comment (Koenkk) [inline src/lib/lumi.ts:2833]: Could you refactor these converters to modernExtend? This replaces the to/fromZigbee and exposes. Example: https://github.com/marthubner/zigbee-herdsman-converters/blob/2ca6323e34117f1686bdac9c98774397c25ece53/src/lib/lumi.ts#L1518

Comment (marthubner) [reply src/lib/lumi.ts]: I tried to do so

File: src/devices/lumi.ts [hunk 2]
```diff
@@ -2521,6 +2536,37 @@ const definitions: Definition[] = [
...
         extend: [lumiZigbeeOTA()],
     },
+    {
+        zigbeeModel: ['lumi.curtain.acn04'],
+        model: 'ZNCLDJ01LM',
+        vendor: 'Aqara',
+        description: 'Curtain controller C3',
+        toZigbee: [lumi.toZigbee.lumi_curtain_limits_calibration, lumi.toZigbee.lumi_curtain_automatic_calibration_ZNCLDJ01LM],
+        exposes: [
+            e.enum('manual_calibration', ea.SET, ['Start', 'End', 'Reset']).withDescription('Calibrate the position limits'),
```
Comment (Koenkk) [inline src/devices/lumi.ts:2546]:
```suggestion
            e.enum('manual_calibration', ea.SET, ['start', 'end', 'reset']).withDescription('Calibrate the position limits'),
```

File: src/devices/lumi.ts [hunk 3]
```diff
@@ -2521,6 +2536,37 @@ const definitions: Definition[] = [
...
+            e
+                .enum('automatic_calibration', ea.SET, ['Calibrate'])
```
Comment (Koenkk) [inline src/devices/lumi.ts:2548]:
```suggestion
                .enum('automatic_calibration', ea.SET, ['calibrate'])
```

File: src/lib/lumi.ts [hunk 4]
```diff
@@ -1536,6 +1537,170 @@ export const lumiModernExtend = {
...
+            attribute: 'windowCoveringMode',
+            description: 'Whether the curtain direction is inverted',
+            access: 'ALL',
+            entityCategory: 'config',
+            ...args,
+        }),
+    lumiCurtainStatus: (args?: Partial<modernExtend.EnumLookupArgs>) =>
+        modernExtend.enumLookup({
+            name: 'status',
+            lookup: {0: 'Closing', 1: 'Opening', 2: 'Stopped', 3: 'Blocked'},
```
Comment (Koenkk) [inline src/lib/lumi.ts:1608]:
```suggestion
            lookup: {0: 'closing', 1: 'opening', 2: 'stopped', 3: 'blocked'},
```

File: src/lib/lumi.ts [hunk 5]
```diff
@@ -1536,6 +1537,170 @@ export const lumiModernExtend = {
...
+            description: 'Current status of the curtain (Opening, Closing, Stopped, Blocked)',
+            zigbeeCommandOptions: {manufacturerCode},
+            access: 'STATE',
+            entityCategory: 'diagnostic',
+            ...args,
+        }),
+    lumiCurtainLastManualOperation: (args?: Partial<modernExtend.EnumLookupArgs>) =>
+        modernExtend.enumLookup({
+            name: 'last_manual_operation',
+            lookup: {1: 'Open', 2: 'Close', 3: 'Stop'},
```
Comment (Koenkk) [inline src/lib/lumi.ts:1620]:
```suggestion
            lookup: {1: 'open', 2: 'close', 3: 'stop'},
```

File: src/lib/lumi.ts [hunk 6]
```diff
@@ -1536,6 +1537,170 @@ export const lumiModernExtend = {
...
+            cluster: 'manuSpecificLumi',
+            attribute: {ID: 0x0425, type: 0x20},
+            description: 'Last triggered manual operation (Open, Close, Stop)',
```
Comment (Koenkk) [inline src/lib/lumi.ts:1623]:
```suggestion
            description: 'Last triggered manual operation',
```

File: src/lib/lumi.ts [hunk 7]
```diff
@@ -1536,6 +1537,170 @@ export const lumiModernExtend = {
...
+            zigbeeCommandOptions: {manufacturerCode},
+            access: 'STATE',
+            unit: 'sec',
+            entityCategory: 'diagnostic',
+            ...args,
+        }),
+    lumiCurtainCalibrationStatus: (args?: Partial<modernExtend.EnumLookupArgs>) =>
+        modernExtend.enumLookup({
+            name: 'calibration_status',
+            lookup: {0: 'Not calibrated', 1: 'Half calibrated', 2: 'Fully calibrated'},
```
Comment (Koenkk) [inline src/lib/lumi.ts:1658]:
```suggestion
            lookup: {0: 'not_calibrated', 1: 'half_calibrated', 2: 'fully_calibrated'},
```

File: src/lib/lumi.ts [hunk 8]
```diff
@@ -1536,6 +1537,170 @@ export const lumiModernExtend = {
...
+            description: 'Indicates if this device is calibrated',
+            zigbeeCommandOptions: {manufacturerCode},
+            access: 'STATE',
+            entityCategory: 'diagnostic',
+            ...args,
+        }),
+    lumiCurtainControlManuSpecific: (args?: Partial<modernExtend.EnumLookupArgs>) =>
+        modernExtend.enumLookup({
+            name: 'control',
+            lookup: {Toggle: 3, Open: 7, Close: 8},
```
Comment (Koenkk) [inline src/lib/lumi.ts:1683]:
```suggestion
            lookup: {toggle: 3, open: 7, close: 8},
```

File: src/lib/lumi.ts [hunk 9]
```diff
@@ -1536,6 +1537,170 @@ export const lumiModernExtend = {
...
+            description: 'manuSpecific curtain control',
+            zigbeeCommandOptions: {manufacturerCode},
+            access: 'ALL',
+            entityCategory: 'config',
+            ...args,
+        }),
+    lumiCurtainIdentifyBeep: (args?: Partial<modernExtend.EnumLookupArgs>) =>
+        modernExtend.enumLookup({
+            name: 'identify_beep',
+            lookup: {Short: 0, '1 sec': 1, '2 sec': 2},
```
Comment (Koenkk) [inline src/lib/lumi.ts:1695]:
```suggestion
            lookup: {short: 0, '1_sec': 1, '2_sec': 2},
```

File: src/lib/lumi.ts [hunk 10]
```diff
@@ -4671,9 +4836,11 @@ export const toZigbee = {
         },
     } satisfies Tz.Converter,
     lumi_curtain_limits_calibration: {
-        key: ['limits_calibration'],
+        key: ['limits_calibration', 'manual_calibration'],
```
Comment (Koenkk) [inline src/lib/lumi.ts:4839]: Why not just use `limits_calibration` instead of adding `manual_calibration`?

File: src/lib/lumi.ts [hunk 11]
```diff
@@ -4706,6 +4873,87 @@ export const toZigbee = {
             }
         },
     } satisfies Tz.Converter,
+    lumi_curtain_automatic_calibration_ZNCLDJ01LM: {
+        key: ['automatic_calibration'],
+        convertSet: async (entity, key, value, meta) => {
+            const NS = 'zhc:lumi:curtain';
```
Comment (Koenkk) [inline src/lib/lumi.ts:4879]: Please define this in the top of the file `const NS = 'zhc:lumi'`

File: src/lib/lumi.ts [hunk 12]
```diff
@@ -4706,6 +4873,87 @@ export const toZigbee = {
...
+            const wait = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
```
Comment (Koenkk) [inline src/lib/lumi.ts:4880]: You can use `utils.sleep()` instead.

File: src/lib/lumi.ts (hunk 11 above)
Comment (marthubner) [reply src/lib/lumi.ts:4879]: Should I use zhc:lumi instead or define zhc:lumi:curtain as another constant? I don't know what NS stands for.

File: src/lib/lumi.ts (hunk 10 above)
Comment (marthubner) [reply src/lib/lumi.ts:4839]: I wanted a distinction between automatic and manual calibration, but never mind, I'll change it

**Comment by Koenkk** [comment]
Thanks!

Could you also submit a picture for the docs? This can be done by:
1. Create a fork by clicking [here](https://github.com/Koenkk/zigbee2mqtt.io/fork)
2. Go to the `public/images/devices` directory, *Add file* -> *Upload files*
3. Upload the files and press *Commit changes*
4. Press *Contribute* -> *Open pull request* -> update title/description -> *Create pull request*

Make sure that:
- The filename is `ZNCLDJ01LM.png`
- The size is 512x512
- The background is transparent (use e.g. [Adobe remove background](https://new.express.adobe.com/tools/remove-background))

**Pull request merged**